Ef code first

TheOverclockedLad opened the pull request.

TheOverclockedLad committed:
Implemented the test module

List of implemented functionalities:
1) CURD for tests and questions
2) Test module which actually conducts tests

TheOverclockedLad committed:
UI changes

1) Modified _Layout.cshtml: added nav links for StartTest, Tests, Questions

2) Replaced "TestId" with "Add to test" and added "Later.." option in the DropDownList for test so as to skip adding the question in a test in Create view of Questions controller

TheOverclockedLad committed:
Working test module

abhaykPere commented on MvcOnlineTest/App_Start/BundleConfig.cs:
```diff
@@ -8,6 +7,11 @@ public class BundleConfig
+                        "~/Scripts/jquery.validate*",
+                        "~/Scripts/jquery.unobtrusive-ajax.js"));
+
             bundles.Add(new ScriptBundle("~/bundles/jquery").Include(
```
scriptBundle does really required multiple

abhaykPere commented on MvcOnlineTest/Controllers/HomeController.cs:
```diff
@@ -13,13 +9,15 @@ public ActionResult Index()
             return View();
         }
 
+        [Authorize(Roles = "admin")]
```
What is reason for roles = "admin"

abhaykPere commented on MvcOnlineTest/Controllers/HomeController.cs:
```diff
@@ -13,13 +9,15 @@ public ActionResult Index()
             return View();
         }
 
+        [Authorize]
```
for contact you added [authorize] attribute but its not for about

abhaykPere commented on MvcOnlineTest/Controllers/LoginsController.cs:
```diff
@@ -0,0 +1,55 @@
+﻿using MvcOnlineTest.Models;
```
are we using this controller

abhaykPere commented on MvcOnlineTest/Controllers/QuestionsController.cs:
```diff
@@ -0,0 +1,144 @@
+        private MvcOnlineTestDb db = new MvcOnlineTestDb();
+
+        // GET: Questions
+        public ActionResult Index()
```
please give meaning full names to all actions

abhaykPere commented on MvcOnlineTest/Controllers/QuestionsController.cs:
```diff
@@ -0,0 +1,144 @@
+        // GET: Questions
+        public ActionResult Index()
+        {
+            var questions = db.Questions.Include(q => q.Test);
```
don't use var

abhaykPere commented on MvcOnlineTest/Controllers/QuestionsController.cs:
```diff
@@ -0,0 +1,144 @@
+        [ValidateAntiForgeryToken]
+        public ActionResult Create([Bind(Include = "QuestionId,TestId,Que,OptionA,OptionB,OptionC,OptionD,Answer")] Question question, bool redirect = false)
+        {
+            if (ModelState.IsValid)
```
use try catch everywhere

abhaykPere commented on MvcOnlineTest/Controllers/StartTestController.cs:
```diff
@@ -0,0 +1,210 @@
+
+        //public PartialViewResult PreviousQuestion(string option)
+        //{
+        //    int index = ((List<Question>)Session["QuestionList"]).IndexOf((Question)Session["Question"]);
```
remove commented code

abhaykPere commented on MvcOnlineTest/Models/Login.cs:
```diff
@@ -0,0 +1,14 @@
+    {
+        [Key]
+        public string username { get; set; }
+
```
validation annotation not added

abhaykPere commented on MvcOnlineTest/Models/Student.cs:
```diff
@@ -0,0 +1,18 @@
+
+        [Required, StringLength(25, MinimumLength = 1, ErrorMessage = "Last name cannot be longer than 25 characters."), Display(Name = "Last Name")]
+        public string LastName { get; set; }
+
```
validation for text input not present

TheOverclockedLad committed:
Refactorings

Deleted unused stuff

Renamed "LoginsController" to "LoginController" and "VM_Leaderboard" to "_Leaderboard"

Added attributes Authorize in Questions and Test controller and HandleError in all controllers

Removed commented code
